### ISSUE TITLE
Notification won't block lxqt-config-monitor

### ISSUE THREAD

**Issue by jfk88** [lxqt/lxqt#1788]
Since I've updated to 0.15.0, `lxqt-config-monitor`pops in the notifications each time I wake up the screen telling me I'm in default mode. I tried to block it in the parameters but that won't work.

**Comment by tsujan** [comment]
Issue template exits!

**Comment by jfk88** [comment]
> Issue template exits!

Thanks, but I promise I looked for the issue beforehand. How is it called? Cannot find it.

**Comment by tsujan** [comment]
Your browser doesn't show a template when you want to make an issue? I'd ignore that if I understood what you said.

Anyway, lxqt-notificationd doesn't block any notification (why should a notification daemon do that?!). Nothing has changed in 0.15.0 in that regard.

I don't know about lxqt-config-monitor's notifications though (don't have any here). @selairi, does lxqt-config-monitor issue notifications?

**Comment by jfk88** [comment]
I'll precise.

Here is the parameter in question (perhaps it's called "ignore"):

![1](https://user-images.githubusercontent.com/25780374/80860979-d28bfb80-8c6b-11ea-8f56-a9115686a0b2.png)

And this is what I got despite ignoring `lxqt-config-monitor`:

![Capture d'écran Deepin_zone de sélection _20200502115218](https://user-images.githubusercontent.com/25780374/80860988-e172ae00-8c6b-11ea-9f6a-07e80f2de1cb.png)

I'm on Arch with LxQt 0.15.0. I also use `autorandr` for `xranrd` profiles.

**Comment by tsujan** [comment]
As the text says, it's "Notifications sans surveillance", which means the listed notifications aren't saved when unattended.

As for `lxqt-config-monitor`, I took a look at the code. Yes, `lxqt-config-monitor` may show a notification but *only when the settings cannot be applied*. Therefore, you might want to check what's wrong with your monitor settings.

You couldn't know that something is wrong if no notification was shown. So, it's useful to you.

**Comment by jfk88** [comment]
> As the text says, it's "Notifications sans surveillance", which means the listed notifications aren't saved when unattended.

OK I misread that indeed.

> As for lxqt-config-monitor, I took a look at the code. Yes, lxqt-config-monitor may show a notification but only when the settings cannot be applied. Therefore, you might want to check what's wrong with your monitor settings.

My guess is that there's a conflict with `autorandr` when the screen wakes up. But as lxqt-config-monitor was [already buggy with kscreen](https://github.com/lxqt/lxqt-config/issues/504) I simply delete the fix in `.xinitrc` to make it useless again. Far from being elegant but this GUI is redundant with `autorandr` and `arandr`.

**Comment by tsujan** [comment]
> But as lxqt-config-monitor was already buggy with kscreen...

A probable problem doesn't mean "buggy". But your system, your choice.

**Comment by jfk88** [comment]
> A probable problem doesn't mean "buggy". But your system, your choice.

Of course but it doesn't follow that I chose to have "probable problems". I mean one of the beauty of getting a Linux-based OS (especially with LxQt which is currently the best environnement in my opinion) is the flexibility it offers. As soon as you have to constraint what you install on your computer because of modules "probable incompatibility" you lose the magic... 

Anyway, thank you very much for kind replies.

**Comment by selairi** [comment]
I think that I could add a checkbox in lxqt-session and the user could disable lxqt-config-monitor. There are lots of users and systems.

I was thinking in Wayland and how to implement lxqt-config-monitor in it. I think that KScreen has a backend to wlroots (Wayfire, Waybox), but in Weston compositor you have to change monitors settings editing a text file, then lxqt-config-monitor is not usable in Weston.

If you use a Wayland compositor not supported be KScreen, this kind of option could be needed.

@tsujan  What do you think?

And does anybody know a way to github warn me when someone writes @selairi in a message? I am getting late to all issues. 😓

Thanks

**Comment by tsujan** [comment]
@selairi 
I think it's a good idea.

> And does anybody know a way to github warn me when someone writes @selairi in a message? I am getting late to all issues. 

Now, I understand why sometimes you don't reply ;)

Go to your GitHub account settings → Notifications and check `Email` (and, probably, `Web and Mobile`) under `Participating` and `Watching`. Also, check `Email` under Github Actions and check the boxes under Automatic watching.

**Comment by selairi** [comment]
> Go to your GitHub account settings → Notifications and check Email (and, probably, Web and Mobile) under Participating and Watching. Also, check Email under Github Actions and check the boxes under Automatic watching.

Done... I am checking the little bell in my account... OMG!! 😮

**Comment by selairi** [comment]
I am thinking where is the best place for that option:
![screenMonitor](https://user-images.githubusercontent.com/1163139/81235325-e97b7680-8ffa-11ea-8f86-5f7f7aa199e6.png)
"Basic Settings" could be the right place.
Is there a better place?

**Comment by tsujan** [comment]
@selairi 
It can be "Detect screen plugging". The place seems good to me but I'm not sure about the text; I think it needs a good tooptip to tell users what this checkbox does. @luis-pereira, @yan12125, what should we write here? Or, maybe, in another place?

**Comment by yan12125** [comment]
Is the option for the following code?

https://github.com/lxqt/lxqt-session/blob/784acb78587c72c7f77dbf8878ef4d51bdffb938/lxqt-session/src/sessionapplication.cpp#L109

If so, maybe the text will be better with "Reload monitor settings when a screen is plugged or unplugged".

> Or, maybe, in another place?

My first intuition is the option is in lxqt-config-monitor, but I'm not sure after seeing the code in lxqt-session...

**Comment by tsujan** [comment]
@yan12125's text is a good explanation. So, I suggest "Detect screen plugging" as the checkbox's text and "Reload monitor settings when a screen is plugged or unplugged" as its tooltip.

**Comment by luis-pereira** [comment]
> @yan12125's text is a good explanation. So, I suggest "Detect screen plugging" as the checkbox's text and "Reload monitor settings when a screen is plugged or unplugged" as its tooltip.

That's good. Very clear.

**Comment by selairi** [comment]
Maybe, 

[https://github.com/lxqt/lxqt-session/blob/784acb78587c72c7f77dbf8878ef4d51bdffb938/lxqt-session/src/sessionapplication.cpp#L109](https://github.com/lxqt/lxqt-session/blob/784acb78587c72c7f77dbf8878ef4d51bdffb938/lxqt-session/src/sessionapplication.cpp#L109)

should be removed from lxqt-session and build a new module in lxqt-config that can be launched from lxqt-session as lxqt-powermanager, lxqt-panel or lxqt-notificationd are run.

**Comment by yan12125** [comment]
> should be removed from lxqt-session and build a new module in lxqt-config

Sounds good to me. Modularity often improves stability. It might bring a little more memory consumption (#1776), though.

**Comment by tsujan** [comment]
> t might bring a little more memory consumption (#1776), though.

https://github.com/lxqt/lxqt/issues/1776 isn't important -- it may even be good in some way (less memory allocations?). I'm only curious to know its cause. Is it specific to Arch (and its derivatives)? If so, how? If not,... etc.

**Comment by selairi** [comment]
@yan12125 and @tsujan Thanks for your answers. I have written a lxqt-monitor module:

[https://github.com/selairi/lxqt-config](https://github.com/selairi/lxqt-config)

The RAM used is:
![mem](https://user-images.githubusercontent.com/1163139/82138630-2f69e300-9822-11ea-862b-124baa6aaa9c.png)

as you can see, the RAM wasted by lxqt-monitor is almost the same as RAM wasted by lxqt-session.

Does a plugin system for lxqt-session waste less RAM? lxqt-panel load the plugins as libraries. Maybe, if lxqt-notificationd, lxqt-policykit-agent, lxqt-powermanagement and lxqt-globalkeysd are loaded by lxqt-session as dynamic libraries, the RAM consumption will be lower. The code looks so simple that I am going to test it. Perhaps, the biggest problem in that plugin system, it would be debug the code.

**Comment by tsujan** [comment]
@selairi 

IMHO, we don't need to worry about a little more or less RAM usage. Make a PR (or PRs) whenever you feel your work is ready.

**Comment by yan12125** [comment]
Agreed with @tsujan. 2.8M does not matter even on systems with small RAM (e.g., Raspberry Pi).

**Comment by stefonarch** [comment]
Stumbled upon this, it's still like that on x11 and kwin_wayland and annoying.
But the solution proposed here doesn't convince me much, as it's not clear what this module is doing, in difference to all others.

Ideally a message should be shown _only_ when it is not possible to apply settings, afaik this is done.

**Comment by tsujan** [comment]
> Stumbled upon this, it's still like that on x11 and kwin_wayland and annoying.

It has never happened to me.

> Ideally a message should be shown only when it is not possible to apply settings, afaik this is done.

→ https://github.com/lxqt/lxqt/issues/1788#issuecomment-622930318 → 2nd paragraph

**Comment by stefonarch** [comment]
> It has never happened to me.

It happens only on a dual monitor setup afaik (or maybe if you configure it to be flipped or inverted too). Also if you remove it from the autostart it starts again when it detects the other monitor after login.

**Comment by tsujan** [comment]
> It happens only…

This important info is missing in the title. Actually, it was missing until now.

**Comment by stefonarch** [comment]
I'm not so sure, as I can see in on x11 with xfm4 and one  monitor too. Maybe because once 2 monitors have been configured?

**Comment by luis-pereira** [comment]
Never happened to me. One monitor only.

**Comment by stefonarch** [comment]
Just logged into x11, it's still there at session start but also when screens are coming back from blanking. Add  "don't show that message again" or only show it once? Better would be showing a message when it fails only (does it fail?).

**Comment by tsujan** [comment]
> does it fail?

Yes. → https://github.com/lxqt/lxqt/issues/1788#issuecomment-622930318
